### PR TITLE
Add Trusted App Metadata to Application management APIs

### PR DESCRIPTION
**Pull request by ImalshaG**
Related issue: https://github.com/wso2/product-is/issues/20487

This PR adds Trusted App Metadata to Application object under advanced configurations section as follows:

```
{
    "advancedConfigurations": {
        "trustedAppConfiguration": {
            "isFIDOTrustedApp": true,
            "isConsentGranted": true,
            "androidPackageName": "com.wso2.mobile.sample1",
            "androidThumbprints": "18:94:0A:DE:63:77:B6:84:43:1E:85:8F:03:CF:8A:14:87:9C:DE:DF:EA:7A:25:53:CD:53:5A:AF:C3:54:A5:56",
            "appleAppId": "APPLETEAMID.com.org.mobile.sample"
        }
    }
}
``` 

Merge after https://github.com/wso2/carbon-identity-framework/pull/5753

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java [hunk 1]
```diff
@@ -160,6 +160,10 @@ public enum ErrorMessage {
         BLOCK_RENAME_APP_NAME_TO_RESERVED_APP_NAME("60516",
                 "Renaming application name to a system reserved name is blocked",
                 "The application name %s is marked as systems reserved application name."),
+        INCORRECT_ANDROID_APP_DETAILS("60516",
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java:163]: error code is used above as well

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml [hunk 2]
```diff
@@ -3142,6 +3144,30 @@ components:
...
+        isConsentGranted:
+          type: boolean
+          description: Decides whether consent is granted for the trusted app.
+          example: false
+        androidPackageName:
+          type: string
+          description: Decides the android package name for the application.
+          example: "com.wso2.mobile.sample"
+        androidThumbprints:
+          type: string
```
Comment (ImalshaG) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml:3164]: make it a list of strings

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml [hunk 3]
```diff
@@ -3142,6 +3144,30 @@ components:
...
+          description: Decides the android thumbprints for the application.
+          example: "18:94:0A:DE:63:77:B6:84:43:1E:85:8F:03:CF:8A:14:87:9C:DE:DF:EA:7A:25:53:CD:53:5A:AF:C3:54:A5:56"
+        appleAppId:
+          type: string
+          description: Decides the apple app id for the application.
+          example: "APPLETEAMID.com.wso2.mobile.sample"
```
Comment (ImalshaG) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml:3170]: Remove wso2

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java [hunk 4]
```diff
@@ -489,6 +493,28 @@ private AdvancedApplicationConfiguration buildAdvancedAppConfiguration(ServicePr
...
+        SpTrustedAppMetadata trustedAppMetadata = serviceProvider.getTrustedAppMetadata();
+        if (trustedAppMetadata == null) {
+            trustedAppMetadata = new SpTrustedAppMetadata();
+        }
+
+        return new TrustedAppConfiguration()
+                .isFIDOTrustedApp(trustedAppMetadata.getIsFidoTrusted())
+                .androidPackageName(trustedAppMetadata.getAndroidPackageName())
+                .androidThumbprints(Arrays.asList(trustedAppMetadata.getAndroidThumbprints()
+                        .split(ATTRIBUTE_SEPARATOR)))
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java:514]: Lets have the thumprints as an array in the model so we can avoid this sort of splitting in multiple places.

**Review by janakamarasena — APPROVED** [review]

**Review by janakamarasena — APPROVED** [review]

**Comment by ImalshaG** [comment]
Needed modifications for integration tests are added by https://github.com/wso2/product-is/pull/20653.  PR builder fails without these changes.

**Pull request merged**